app/models/job.ts

```diff
@@ -408,6 +416,62 @@ export class Job extends Record {
     return terminalStates.includes(this.status);
   }
 
+  /**
+   * Checks whether sharing of this job is restricted by any EULAs for
+   * any collection used by this job.
+   * Defaults to true if any collection does not have the harmony.has-eula tag
+   * associated with it.
+   * @param accessToken - the token to make the request with
+   * @returns true or false
+   */
+  async collectionsHaveEulaRestriction(accessToken: string): Promise<boolean> {
+    const cmrCollections = await getCollectionsByIds(this.collectionIds, accessToken, 'harmony.has-eula');
```

Comment:
We should pull out `harmony.has-eula` to a named constant.

Here's some examples of other groups tags (not sure if we might want to structure things so we have a single JSON tag like). There are pros and cons to having a single harmony tag vs. potentially many. A single tag makes it easy to pull back all harmony information, but would require updates to the tags to include all of the harmony info:
```
harmony.collection_info
```
  data: {
    has_eula: false
  }
}
```
edsc.extra.serverless.collection_capabilities: {
  data: {
    cloud_cover: false,
    day_night_flag: false,
    granule_online_access_flag: false,
    orbit_calculated_spatial_domains: false,
    updated_at: "2021-07-20T12:00:35.493Z"
  }
}
```

```
opensearch.granule.osdd: {
  data: "http://fedeo.esa.int/opensearch/description.xml?parentIdentifier=EOP:ESA:FEDEO:COLLECTIONS:CryoSat.products"
}
```

Reply:
Those are some good options. I think we can also return tags by wildcard -- e.g. harmony*. I made the key a constant for now: https://github.com/nasa/harmony/commit/41d36f98e3cfed748c8dee0433bb0f1c5c824070